analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.Nullable.cs

```diff
@@ -82,4 +84,26 @@ public void Parenthesis(bool condition)
             o.ToString();                            // Noncompliant The ! only suppresses the warning on the expression, but is not applied to any inner variables.
         }
     }
+
+    public class NullableAnnotations
+    {
+        public void DebugAssert()
+        {
+            object? o = null;
+            Debug.Assert(o != null);
+            o.ToString();  // Compliant. Debug.Assert is annotated in .NetStandard with Debug.Assert([DoesNotReturnIf(false)]bool condition)
+        }
+
+        public void NotNullWhenFalse()
+        {
+            string? s = null;
+            if (!IsNullOrEmpty(s))
+            {
+                s.ToString();  // Compliant
+            }
+        }
+
+        private static bool IsNullOrEmpty([NotNullWhen(false)] string? s)
```

Comment:
To avoid confusiton with `string.IsNullOrEmpty`
```suggestion
        private static bool CustomNotNullAssertion([NotNullWhen(false)] string? s)
```